src/validator.cpp

```diff
@@ -1183,145 +1156,89 @@ bool Validator::ValidatorImpl::unitsAreEquivalent(const ModelPtr &model,
-    std::string myPre;
-    std::string myId;
-    double myExp;
-    double myMult;
-    double m;
-    std::map<std::string, double> myBase;
-    libcellml::UnitsPtr u = std::make_shared<libcellml::Units>();
-
-    if (model->hasUnits(uName)) {
-        u = model->units(uName);
-        if (!u->isBaseUnit()) {
-            for (size_t i = 0; i < u->unitCount(); ++i) {
-                u->unitAttributes(i, myRef, myPre, myExp, m, myId);
-                myMult = std::log10(m);
-                if (!isStandardUnitName(myRef)) {
-                    incrementBaseUnitCount(model, unitMap, multMap, myRef, myExp * uExp, logMult + myMult * uExp + standardPrefixList.at(myPre) * uExp);
-                } else {
-                    myBase = standardUnitsList.at(myRef);
-                    for (const auto &iter : myBase) {
-                        unitMap.at(iter.first) += iter.second * myExp * uExp;
-                    }
-                    multMap += logMult + (standardMultiplierList.at(myRef) + myMult + standardPrefixList.at(myPre)) * uExp;
-                }
-            }
-        } else if (unitMap.find(uName) == unitMap.end()) {
-            unitMap.emplace(std::pair<std::string, double>(uName, uExp));
-            multMap += logMult;
-        }
-    } else if (isStandardUnitName(uName)) {
-        myBase = standardUnitsList.at(uName);
-        for (const auto &iter : myBase) {
-            unitMap.at(iter.first) += iter.second * uExp;
-        }
-        multMap += logMult + standardMultiplierList.at(uName);
-    }
-}
-
-void Validator::ValidatorImpl::decrementBaseUnitCount(const ModelPtr &model,
-                                                      std::map<std::string, double> &unitMap,
-                                                      double &multMap,
-                                                      const std::string &uName,
-                                                      const double &uExp,
-                                                      const double &logMult)
+void Validator::ValidatorImpl::updateBaseUnitCount(const ModelPtr &model,
+                                                   std::map<std::string, double> &unitMap,
+                                                   const std::string &uName,
+                                                   double uExp, double logMult,
+                                                   int direction)
 {
-    std::string myRef;
-    std::string myPre;
-    std::string myId;
-    double myExp;
-    double myMult;
-    double m;
-    std::map<std::string, double> myBase;
-    libcellml::UnitsPtr u = std::make_shared<libcellml::Units>();
-
     if (model->hasUnits(uName)) {
-        u = model->units(uName);
+        libcellml::UnitsPtr u = model->units(uName);
         if (!u->isBaseUnit()) {
+            std::string myRef;
+            std::string myPre;
+            std::string myId;
+            double myExp;
+            double myMult;
+            double m;
```

Comment:
Yes, I don't like the `m` variable it seems it also means the `myMult` is also a bit disingenuous.

Reply:
All fine now.